sqlmesh/core/model/definition.py

```diff
@@ -645,15 +649,15 @@ def is_breaking_change(self, previous: Model) -> t.Optional[bool]:
         """
         raise NotImplementedError
 
-    @property
-    def data_hash(self) -> str:
+    def data_hash(self, additional_fields: t.Optional[t.List[str]] = None) -> str:
```

Comment:
why are you doing additional fields and not just make it first class?

Reply:
Model generates the fingerprint and `physical_schema` is a Snapshot concept and not a model concept. So rather than introducing an unrelated concept to Model I instead allowed Snapshot to pass in additional fields into the data hash so it could share it's context in cases like this.

Reply:
the fingerprint used to get generated in snapshot and it recently got moved to model...

Reply:
i don't like this because now you need to call data_hash with special arguments to get the right hash, i think we need to move physical_schema into the model, or make this purely in the snapshot fingerprint level, meaning it's an arg to the snapshotfingerprint

Reply:
@izeigerman thoughts?

Reply:
My original contention in the PR was that the physical schema looked like it needed to be part of the Model. In the same way that dialect can be derived from the model_defaults section. My 2 cents. 


Models should have a physical schema and virtual schema config. You can set it in the MODEL properties. At a project level, you can use model_defaults.physical_schema as either a string or map. If it is a map, then it is a map of HashMap<SourceSchema, TargetSchema> where source schema is basically the schema part of the model name and target schema is whatever you want. In this way you can leave the default behavior by doing nothing, force all snapshots into a single schema by supplying a string, or use a mapping to route models to specific schemas, possibly with a `__default__` key in the mapping.

Reply:
Talked to @izeigerman. Going to add it to Model but make it a root level config property since we don't want users overriding this on a per-model basis.